plugin/pkg/skplug/dispatcher/dispatcher.go

```diff
@@ -25,10 +25,23 @@ var _ skplug.Plugin = &dispatcher{}
 var _ Dispatcher = &dispatcher{}
 
 func (d *dispatcher) Event(partition string, time int64, typ proto.EventType, object skplug.Object) error {
-	for _, pluginServer := range d.capabilityToPlugins[proto.Capability_EVENT] {
-		err := (*pluginServer).Event(partition, time, typ, object)
-		if err != nil {
-			return err
+	switch o := object.(type) {
+	case *skplug.Autoscaler:
```

Comment:
This works.  Perhaps add a TODO to separate partition creation from events.  Because there is clearly some mismatch based on the code you had to write to make this work.

Reply:
Added it.